components/core/src/clp_s/search/ast/FilterExpr.hpp

```diff
@@ -59,32 +43,55 @@ class FilterExpr : public Expression {
      * @param parent parent this expression is attached to
      * @return newly created Or expression
      */
-    static std::shared_ptr<Expression> create(
+    [[nodiscard]] static auto create(
             std::shared_ptr<ColumnDescriptor>& column,
             FilterOperation op,
             std::shared_ptr<Literal>& operand,
             bool inverted = false,
             Expression* parent = nullptr
-    );
+    ) -> std::shared_ptr<Expression>;
 
     /**
      * Helper function to turn FilterOperation into string for printing
      * @param op the operation we want to convert to string
      * @return a string representing the operation
      */
-    static std::string op_type_str(FilterOperation op);
+    [[nodiscard]] static auto op_type_str(FilterOperation op) -> std::string;
+
+    auto operator=(FilterExpr const&) -> FilterExpr& = delete;
+
+    FilterExpr(FilterExpr&&) = delete;
+    auto operator=(FilterExpr&&) -> FilterExpr& = delete;
+
+    ~FilterExpr() override = default;
 
     // Methods inherited from Value
-    void print() const override;
+    auto print() const -> void override;
 
     // Methods inherited from Expression
-    bool has_only_expression_operands() override { return false; }
+    [[nodiscard]] auto has_only_expression_operands() -> bool override { return false; }
 
-    std::shared_ptr<Expression> copy() const override;
+    [[nodiscard]] auto copy() const -> std::shared_ptr<Expression> override;
 
-private:
-    FilterOperation m_op;
+    // Methods
+    /**
+     * @return FilterOperation this Filter performs
```

Comment:
```suggestion
     * @return The FilterOperation performed by this Filter.
```
Just while we're touching this comment anyway.